osquery/tables/networking/posix/lldp_neighbors.cpp

```diff
@@ -0,0 +1,421 @@
+  if (lldpctl_atom_get_int(chassis_, lldpctl_k_chassis_cap_available) & mask) {
+    row_[availCol] = "1";
+    row_[enabledCol] =
+        (lldpctl_atom_get_int(chassis_, lldpctl_k_chassis_cap_enabled) & mask)
+            ? "1"
+            : "0";
+  }
+}
+
+inline void LLDPNeighbor::getPMDAutoNeg(long int autonegAdversised,
+                                        long int hdMask,
+                                        long int fdMask,
+                                        std::string hdCol,
+                                        std::string fdCol) {
+  row_[hdCol] = ((autonegAdversised & hdMask) ? "1" : "0");
+  row_[fdCol] = ((autonegAdversised & fdMask) ? "1" : "0");
+}
+
+void LLDPNeighbor::getPIDs() {
+  lldpctl_atom_t* pids = lldpctl_atom_get(port_, lldpctl_k_port_pis);
+  row_["pids"] = commaDelimitedStr(pids, lldpctl_k_pi_id);
+
+  lldpctl_atom_dec_ref(pids);
+}
+
+void LLDPNeighbor::getPPVIDs() {
+  lldpctl_atom_t* ppvids = lldpctl_atom_get(port_, lldpctl_k_port_ppvids);
+  if (!ppvids) {
+    return;
+  }
+
+  std::string supported;
+  std::string enabled;
+
+  lldpctl_atom_t* ppvid;
+  lldpctl_atom_foreach(ppvids, ppvid) {
+    long int status = lldpctl_atom_get_int(ppvid, lldpctl_k_ppvid_status);
+    long int id = lldpctl_atom_get_int(ppvid, lldpctl_k_ppvid_id);
+
+    if (status > 0 && id > 0) {
+      std::string name = getAtomStr(ppvid, lldpctl_k_ppvid_id);
+
+      if (status & LLDP_PPVID_CAP_SUPPORTED) {
+        supported = supported + name + ",";
+      }
+
+      if (status & LLDP_PPVID_CAP_ENABLED) {
+        enabled = enabled + name + ",";
+      }
+    }
+  }
+
+  if (supported.size() > 0) {
+    supported.pop_back();
+  }
+  row_["ppvids_supported"] = supported;
+
+  if (enabled.size() > 0) {
+    enabled.pop_back();
+  }
+  row_["ppvids_enabled"] = enabled;
+
+  lldpctl_atom_dec_ref(ppvids);
+}
+
+void LLDPNeighbor::getMEDCap() {
+  long int availableCap =
+      lldpctl_atom_get_int(chassis_, lldpctl_k_chassis_med_cap);
+  if (availableCap < 0) {
+    return;
+  }
+
+  row_["med_capability_capabilities"] =
+      (availableCap & LLDP_MED_CAP_CAP) ? "1" : "0";
+  row_["med_capability_policy"] =
+      (availableCap & LLDP_MED_CAP_POLICY) ? "1" : "0";
+  row_["med_capability_location"] =
+      (availableCap & LLDP_MED_CAP_LOCATION) ? "1" : "0";
+  row_["med_capability_mdi_pse"] =
+      (availableCap & LLDP_MED_CAP_MDI_PSE) ? "1" : "0";
+  row_["med_capability_mdi_pd"] =
+      (availableCap & LLDP_MED_CAP_MDI_PD) ? "1" : "0";
+  row_["med_capability_inventory"] =
+      (availableCap & LLDP_MED_CAP_IV) ? "1" : "0";
+}
+
+void LLDPNeighbor::getMED() {
+  if (lldpctl_atom_get_int(chassis_, lldpctl_k_chassis_med_type) <= 0) {
+    return;
+  }
+
+  row_["med_device_type"] = getAtomStr(chassis_, lldpctl_k_chassis_med_type);
+  // Capabilities status
+  getMEDCap();
+
+  // Policy Capabilities
+  // TODO add MED policy tags and defined/undefined tags
```

Comment:
Is it possible to not have TODOs? What scope of additional work are we talking about?